src/Core/src/Platform/Windows/KeyboardAcceleratorExtensions.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+using System;
+using System.Collections.Generic;
+using Windows.System;
+
+namespace Microsoft.Maui.Platform
+{
+	/// <summary>
+	/// Group of helper extension methods related to KeyboardAccelerators.
+	/// </summary>
+	public static class KeyboardAcceleratorExtensions
+	{
+		/// <summary>
+		/// Updates the collection of KeyboardAccelerators used by a MenuFlyoutItemBase.
+		/// </summary>
+		/// <param name="platformView">The platform view, of type MenuFlyoutItemBase.</param>
+		/// <param name="menuFlyoutItem">The abstract MenuFlyoutItem with all the necessary information.</param>
+		public static void UpdateAccelerator(this MenuFlyoutItemBase platformView, IMenuFlyoutItem menuFlyoutItem)
+		{
+			var keyboardAccelerators = menuFlyoutItem.Accelerators?.ToPlatform();
+
+			if (keyboardAccelerators is null)
+				return;
+
+			foreach (var keyboardAccelerator in keyboardAccelerators)
+				platformView.KeyboardAccelerators.Add(keyboardAccelerator);
+		}
+
+		/// <summary>
+		/// Converts a list of IAccelerator to a list of KeyboardAccelerator.
+		/// A KeyboardAccelerator represents a keyboard shortcut (or accelerator) that lets a user perform an action using the keyboard instead
+		/// of navigating the app UI (directly or through access keys). 
+		/// </summary>
+		/// <param name="accelerators">List of IAccelerator</param>
+		/// <returns>List of KeyboardAccelerator</returns>
+		public static IList<KeyboardAccelerator>? ToPlatform(this IReadOnlyList<IAccelerator> accelerators)
+		{
+			if (accelerators is null)
+				return null;
+
+			List<KeyboardAccelerator> result = new List<KeyboardAccelerator>();
+
+			foreach (var accelerator in accelerators)
+			{
+				var keyboardAccelerators = accelerator.ToPlatform();
+
+				if (keyboardAccelerators is not null)
+					result.AddRange(keyboardAccelerators);
+			}
+
+			return result;
+		}
+
+		// Single key (A, Delete, F2, Spacebar, Esc, Multimedia Key) accelerators and multi-key
+		// accelerators (Ctrl+Shift+M) are supported.
+		// Gamepad virtual keys are not supported.
+		public static IList<KeyboardAccelerator>? ToPlatform(this IAccelerator accelerator)
+		{
+			if (accelerator is null)
+				return null;
+
+			List<KeyboardAccelerator> result = new List<KeyboardAccelerator>();
+
+			var key = accelerator.Key;
+			var modifiers = accelerator.Modifiers;
+
+			var keyboardAccelerator = new KeyboardAccelerator();
+			keyboardAccelerator.Key = key.ToVirtualKey();
+			if (modifiers is not null)
```

Comment:
Making modifiers to be an empty array rather than nullable will remove this type of checks.